src/main/java/org/opensearch/knn/index/util/Faiss.java

```diff
@@ -246,7 +247,7 @@ class Faiss extends NativeLibrary {
                     ).addParameter(METHOD_PARAMETER_M, "", "").addParameter(METHOD_ENCODER_PARAMETER, ",", "").build())
                 )
                 .build()
-        ).addSpaces(SpaceType.L2, SpaceType.INNER_PRODUCT).build(),
+        ).addSpaces(SpaceType.UNDEFINED, SpaceType.HAMMING_BIT, SpaceType.L2, SpaceType.INNER_PRODUCT).build(),
```

Comment:
When user does not define space type, it is set as UNDEFINED first. It is needed so that we can have different default space type based on data_type in the later phase.